File: app/src/main/java/net/mbonnin/arcanetracker/StopServiceBroadcastReceiver.java
package net.mbonnin.arcanetracker;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;

import com.esotericsoftware.kryo.util.Util;

/**
 * Created by martin on 10/24/16.
 */

public class StopServiceBroadcastReceiver extends BroadcastReceiver {
    public static final String ACTION = "net.mbonnin.arcanetracker.StopServiceBroadcastReceiver";

    public static Intent getIntent() {
        Intent intent = new Intent();
        intent.setAction(ACTION);

        return intent;
    }

    public static void init() {
        IntentFilter filter = new IntentFilter();
        filter.addAction(ACTION);
        ArcaneTrackerApplication.getContext().registerReceiver(new StopServiceBroadcastReceiver(), filter);
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        Utils.exitApp();
    }
}

File: app/src/main/java/net/mbonnin/arcanetracker/StopServiceBroadcastReceiver.kt
package net.mbonnin.arcanetracker

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter

/**
 * Created by martin on 10/24/16.
 */

class StopServiceBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Utils.exitApp()
    }

    companion object {
        val ACTION = "net.mbonnin.arcanetracker.StopServiceBroadcastReceiver"

        val intent: Intent
            get() {
                val intent = Intent()
                intent.action = ACTION

                return intent
            }

        fun init() {
            val filter = IntentFilter()
            filter.addAction(ACTION)
            ArcaneTrackerApplication.context.registerReceiver(StopServiceBroadcastReceiver(), filter)
        }
    }
}
